### Subpunctul a)

In [14]:
import random

def simulate_game():
    urna = ['rosie'] * 3 + ['albastra'] * 4 + ['neagra'] * 2

    zar = random.randint(1, 6)

    if zar in [2, 3, 5]:
        urna.append('neagra')
    elif zar == 6:
        urna.append('rosie')
    else:
        urna.append('albastra')

    bila_extrasa = random.choice(urna)

    return bila_extrasa

bila_extrasa = simulate_game()
print(f'Bila extrasă este: {bila_extrasa}')

Bila extrasă este: rosie


### Subpunctul b)

In [15]:
def simulate_exp(num_sim):
    nr_bile_rosii = 0

    for _ in range(num_sim):
        bila_extrasa = simulate_game()

        if bila_extrasa == 'rosie':
            nr_bile_rosii += 1

    prob_estimata = nr_bile_rosii / num_sim
    return prob_estimata

prob_estimata_bila_rosie = simulate_exp(10000)
print(f'P(rosie) = {prob_estimata_bila_rosie:.4f}')

P(rosie) = 0.3189


### Bonus

Calculam probabilitatea teoretica a evenimentului si o comparam cu cea obtinuta mai sus.

### 1. Urna inițială:

$$
\text{Total bile inițiale} = 3 \, (\text{roșii}) + 4 \, (\text{albastre}) + 2 \, (\text{negre}) = 9 \, \text{bile}
$$

### 2. Evenimentele posibile la aruncarea zarului:
- Dacă aruncăm un **număr prim** (2, 3, 5), adăugăm o **bilă neagră** în urnă.
- Dacă aruncăm un **6**, adăugăm o **bilă roșie** în urnă.
- În alte cazuri (1 și 4), adăugăm o **bilă albastră**.

### 3. Cazul 1: Obtinem un numar prim
- Adăugăm o bilă neagră în urnă. Urna va conține:
$$
3 \, (\text{roșii}) + 4 \, (\text{albastre}) + 3 \, (\text{negre}) = 10 \, \text{bile}
$$
- Probabilitatea de a extrage o bilă roșie după această modificare:
$$
P(\text{roșie | număr prim}) = \frac{3}{10}
$$
- Probabilitatea de a obține un număr prim la aruncarea zarului este:
$$
P(\text{număr prim}) = \frac{3}{6}
$$

### 4. Cazul 2: Obtinem 6
- Adăugăm o bilă roșie în urnă. Urna va conține:
$$
4 \, (\text{roșii}) + 4 \, (\text{albastre}) + 2 \, (\text{negre}) = 10 \, \text{bile}
$$
- Probabilitatea de a extrage o bilă roșie după această modificare:
$$
P(\text{roșie | 6}) = \frac{4}{10}
$$
- Probabilitatea de a obține un 6 la aruncarea zarului este:
$$
P(\text{6}) = \frac{1}{6}
$$

### 5. Cazul 3: Obtinem 1 sau 4 (restul cazurilor)
- Adăugăm o bilă albastră în urnă. Urna va conține:
$$
3 \, (\text{roșii}) + 5 \, (\text{albastre}) + 2 \, (\text{negre}) = 10 \, \text{bile}
$$
- Probabilitatea de a extrage o bilă roșie după această modificare:
$$
P(\text{roșie | 1 sau 4}) = \frac{3}{10}
$$
- Probabilitatea de a obține 1 sau 4 la aruncarea zarului este:
$$
P(\text{1 sau 4}) = \frac{2}{6}
$$

### 6. Probabilitatea totală teoretică
Probabilitatea totală teoretică de a extrage o bilă roșie este suma ponderată a probabilităților din cele 3 cazuri:
$$
P(\text{roșie}) = P(\text{roșie | număr prim}) \cdot P(\text{număr prim}) + P(\text{roșie | 6}) \cdot P(\text{6}) + P(\text{roșie | 1 sau 4}) \cdot P(\text{1 sau 4}) = \frac{3}{10} \cdot \frac{3}{6} + \frac{4}{10} \cdot \frac{1}{6} + \frac{3}{10} \cdot \frac{2}{6} = \frac{19}{60} = 0.3166
$$

Probabilitatea calculata anterior experimental:
$$
P(\text{roșie}) = 0.3189
$$
